### ISSUE TITLE
Failed to activate the atom-overtype-mode package

### ISSUE THREAD

**Issue by skewty** [brunetton/atom-overtype-mode#8]
[Enter steps to reproduce below:]

1. ...
2. ...

**Atom Version**: 1.0.0
**System**: Microsoft Windows 8.1 Enterprise
**Thrown From**: [atom-overtype-mode](https://github.com/brunetton/atom-overtype-mode) package, v0.2.3

### Stack Trace

Failed to activate the atom-overtype-mode package

```
At Cannot read property 'command' of undefined

TypeError: Cannot read property 'command' of undefined
  at OvertypeMode.activate (C:\Users\spalmer1009\.atom\packages\atom-overtype-mode\lib\atom-overtype-mode.coffee:11:23)
  at Package.module.exports.Package.activateNow (C:\Users\spalmer1009\AppData\Local\atom\app-1.0.0\resources\app.asar\src\package.js:240:19)
  at C:\Users\spalmer1009\AppData\Local\atom\app-1.0.0\resources\app.asar\src\package.js:772:25
  at Emitter.module.exports.Emitter.emit (C:\Users\spalmer1009\AppData\Local\atom\app-1.0.0\resources\app.asar\node_modules\event-kit\lib\emitter.js:82:11)
  at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (C:\Users\spalmer1009\AppData\Local\atom\app-1.0.0\resources\app.asar\src\command-registry.js:222:20)
  at C:\Users\spalmer1009\AppData\Local\atom\app-1.0.0\resources\app.asar\src\command-registry.js:3:61
  at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (C:\Users\spalmer1009\AppData\Local\atom\app-1.0.0\resources\app.asar\node_modules\atom-keymap\lib\keymap-manager.js:520:16)
  at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (C:\Users\spalmer1009\AppData\Local\atom\app-1.0.0\resources\app.asar\node_modules\atom-keymap\lib\keymap-manager.js:355:22)
  at HTMLDocument.module.exports.WindowEventHandler.onKeydown (C:\Users\spalmer1009\AppData\Local\atom\app-1.0.0\resources\app.asar\src\window-event-handler.js:180:20)

```

### Commands

```
     -0:03.4.0 overtype-mode:toggle (atom-text-editor.editor.is-focused)
```

### Config

```json
{
  "core": {
    "disabledPackages": [
      "autocomplete-plus-python-jedi",
      "linter-flake8"
    ],
    "projectHome": "C:\\Development",
    "themes": [
      "one-dark-ui",
      "solarized-dark-syntax"
    ]
  }
}
```

### Installed Packages

```coffee
# User
atom-html-preview, v0.1.8
atom-overtype-mode, v0.2.3
autocomplete-python, v0.5.1
linter, v1.1.0
Sublime-Style-Column-Selection, v1.3.0

# Dev
No dev packages
```

**Comment by Starsam80** [comment]
I got the same error

[Enter steps to reproduce below:]

1. Press `Insert` while in a text document

**Atom Version**: 1.0.2
**System**: Microsoft Windows 10 Pro Insider Preview
**Thrown From**: [atom-overtype-mode](https://github.com/brunetton/atom-overtype-mode) package, v0.2.3


### Stack Trace

Failed to activate the atom-overtype-mode package

```
At Cannot read property 'command' of undefined

TypeError: Cannot read property 'command' of undefined
    at OvertypeMode.activate (C:\Users\Starsam80\.atom\packages\atom-overtype-mode\lib\atom-overtype-mode.coffee:14:32)
    at Package.module.exports.Package.activateNow (C:\Users\Starsam80\AppData\Local\atom\app-1.0.2\resources\app.asar\src\package.js:240:19)
    at C:\Users\Starsam80\AppData\Local\atom\app-1.0.2\resources\app.asar\src\package.js:772:25
    at Emitter.module.exports.Emitter.emit (C:\Users\Starsam80\AppData\Local\atom\app-1.0.2\resources\app.asar\node_modules\event-kit\lib\emitter.js:82:11)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (C:\Users\Starsam80\AppData\Local\atom\app-1.0.2\resources\app.asar\src\command-registry.js:222:20)
    at C:\Users\Starsam80\AppData\Local\atom\app-1.0.2\resources\app.asar\src\command-registry.js:3:61
    at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (C:\Users\Starsam80\AppData\Local\atom\app-1.0.2\resources\app.asar\node_modules\atom-keymap\lib\keymap-manager.js:520:16)
    at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (C:\Users\Starsam80\AppData\Local\atom\app-1.0.2\resources\app.asar\node_modules\atom-keymap\lib\keymap-manager.js:355:22)
    at HTMLDocument.module.exports.WindowEventHandler.onKeydown (C:\Users\Starsam80\AppData\Local\atom\app-1.0.2\resources\app.asar\src\window-event-handler.js:177:20)
```

### Commands

```
     -0:03.3.0 overtype-mode:toggle (atom-text-editor.editor.is-focused)
```

### Config

```json
{
  "core": {
    "themes": [
      "atom-material-ui",
      "atom-material-syntax"
    ]
  }
}
```

### Installed Packages

```coffee
# User
atom-material-syntax, v0.2.1
atom-material-ui, v0.5.2
atom-overtype-mode, v0.2.3
color-picker, v2.0.10
minimap, v4.11.2
pigments, v0.8.4

# Dev
No dev packages
```

**Comment by csoroz** [comment]
[Enter steps to reproduce below:]

1. Press Insert key

**Atom Version**: 1.0.4
**System**: Microsoft Windows 8.1 Pro
**Thrown From**: [atom-overtype-mode](https://github.com/brunetton/atom-overtype-mode) package, v0.2.3


### Stack Trace

Failed to activate the atom-overtype-mode package

```
At Cannot read property 'command' of undefined

TypeError: Cannot read property 'command' of undefined
    at OvertypeMode.activate (C:\Users\csuberviola\.atom\packages\atom-overtype-mode\lib\atom-overtype-mode.coffee:14:32)
    at Package.module.exports.Package.activateNow (C:\Users\csuberviola\AppData\Local\atom\app-1.0.4\resources\app.asar\src\package.js:243:19)
    at C:\Users\csuberviola\AppData\Local\atom\app-1.0.4\resources\app.asar\src\package.js:836:25
    at Emitter.module.exports.Emitter.emit (C:\Users\csuberviola\AppData\Local\atom\app-1.0.4\resources\app.asar\node_modules\event-kit\lib\emitter.js:82:11)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (C:\Users\csuberviola\AppData\Local\atom\app-1.0.4\resources\app.asar\src\command-registry.js:222:20)
    at C:\Users\csuberviola\AppData\Local\atom\app-1.0.4\resources\app.asar\src\command-registry.js:3:61
    at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (C:\Users\csuberviola\AppData\Local\atom\app-1.0.4\resources\app.asar\node_modules\atom-keymap\lib\keymap-manager.js:524:16)
    at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (C:\Users\csuberviola\AppData\Local\atom\app-1.0.4\resources\app.asar\node_modules\atom-keymap\lib\keymap-manager.js:347:22)
    at HTMLDocument.module.exports.WindowEventHandler.onKeydown (C:\Users\csuberviola\AppData\Local\atom\app-1.0.4\resources\app.asar\src\window-event-handler.js:177:20)
```

### Commands

```
     -0:03.4.0 overtype-mode:toggle (atom-text-editor.editor.is-focused)
```

### Config

```json
{
  "core": {}
}
```

### Installed Packages

```coffee
# User
atom-overtype-mode, v0.2.3
language-haskell, v1.2.1

# Dev
No dev packages
```

**Comment by rafalcieslak** [comment]
[Enter steps to reproduce below:]

1. Install atom-overtype-mode
2. Press `Insert` key

**Atom Version**: 1.0.7
**System**: Windows 7
**Thrown From**: [atom-overtype-mode](https://github.com/brunetton/atom-overtype-mode) package, v0.2.3


### Stack Trace

Failed to activate the atom-overtype-mode package

```
At Cannot read property 'command' of undefined

TypeError: Cannot read property 'command' of undefined
    at OvertypeMode.activate (C:\Users\praktykant\.atom\packages\atom-overtype-mode\lib\atom-overtype-mode.coffee:14:32)
    at Package.module.exports.Package.activateNow (C:\Users\praktykant\AppData\Local\atom\app-1.0.7\resources\app.asar\src\package.js:243:19)
    at C:\Users\praktykant\AppData\Local\atom\app-1.0.7\resources\app.asar\src\package.js:836:25
    at Emitter.module.exports.Emitter.emit (C:\Users\praktykant\AppData\Local\atom\app-1.0.7\resources\app.asar\node_modules\event-kit\lib\emitter.js:82:11)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (C:\Users\praktykant\AppData\Local\atom\app-1.0.7\resources\app.asar\src\command-registry.js:222:20)
    at C:\Users\praktykant\AppData\Local\atom\app-1.0.7\resources\app.asar\src\command-registry.js:3:61
    at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (C:\Users\praktykant\AppData\Local\atom\app-1.0.7\resources\app.asar\node_modules\atom-keymap\lib\keymap-manager.js:524:16)
    at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (C:\Users\praktykant\AppData\Local\atom\app-1.0.7\resources\app.asar\node_modules\atom-keymap\lib\keymap-manager.js:347:22)
    at HTMLDocument.module.exports.WindowEventHandler.onKeydown (C:\Users\praktykant\AppData\Local\atom\app-1.0.7\resources\app.asar\src\window-event-handler.js:177:20)
```

### Commands

```
  2x -2:34.1.0 core:confirm (atom-text-editor.editor.mini.is-focused)
  2x -0:19.7.0 core:move-left (atom-text-editor.editor.is-focused)
     -0:19 overtype-mode:toggle (atom-text-editor.editor.is-focused)
     -0:17.6.0 core:backspace (atom-text-editor.editor.is-focused)
```

### Config

```json
{
  "core": {}
}
```

### Installed Packages

```coffee
# User
atom-overtype-mode, v0.2.3
hidpi, v0.7.0
language-cmake, v0.1.4
language-supercollider, v0.2.0
minimap, v4.12.3

# Dev
No dev packages
```

**Comment by cganterh** [comment]
[Enter steps to reproduce below:]

1. ...
2. ...

**Atom Version**: 1.0.5
**System**: Ubuntu 15.04
**Thrown From**: [atom-overtype-mode](https://github.com/brunetton/atom-overtype-mode) package, v0.2.3


### Stack Trace

Failed to activate the atom-overtype-mode package

```
At Cannot read property 'command' of undefined

TypeError: Cannot read property 'command' of undefined
    at OvertypeMode.activate (/home/cristobal/.atom/packages/atom-overtype-mode/lib/atom-overtype-mode.coffee:14:32)
    at Package.module.exports.Package.activateNow (/usr/share/atom/resources/app.asar/src/package.js:243:19)
    at /usr/share/atom/resources/app.asar/src/package.js:836:25
    at Emitter.module.exports.Emitter.emit (/usr/share/atom/resources/app.asar/node_modules/event-kit/lib/emitter.js:82:11)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/usr/share/atom/resources/app.asar/src/command-registry.js:222:20)
    at /usr/share/atom/resources/app.asar/src/command-registry.js:3:61
    at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (/usr/share/atom/resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:524:16)
    at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (/usr/share/atom/resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:347:22)
    at HTMLDocument.module.exports.WindowEventHandler.onKeydown (/usr/share/atom/resources/app.asar/src/window-event-handler.js:177:20)
```

### Commands

```
     -0:00.8.0 overtype-mode:toggle (atom-text-editor.editor.is-focused)
```

### Config

```json
{
  "core": {
    "themes": [
      "one-light-ui",
      "one-light-syntax"
    ],
    "autoHideMenuBar": true,
    "destroyEmptyPanes": false
  }
}
```

### Installed Packages

```coffee
# User
atom-overtype-mode, v0.2.3
autocomplete-python, v0.5.3
linter, v1.4.3
linter-flake8, v1.6.1

# Dev
No dev packages
```

**Comment by CheeseSucker** [comment]
I have created a fixed version of this package. See https://atom.io/packages/atom-overtype-mode-fixed

**Comment by brunetton** [comment]
Hi,
I'm very busy since some weeks, and had no time to have a look on pull requests ! 
@CheeseSucker , Thanks. I added to you collaborators on the project, so you'll be able to push.
What do you think about ? I don't like the idea of having multiple projects doing the same thing; its confuses collaborators, its confuses users.

**Comment by CheeseSucker** [comment]
Hi @brunetton,
Thanks for adding me. I fully agree with you! It seemed like this project had died and there were no alternative packages.

I will unpublish my package and update this one instead. :+1:

**Comment by CheeseSucker** [comment]
Fixed in v0.3.0.
